david-martin/multi-cluster-traffic-controller
# #85: Reconcile Gateway from watching cluster secret

## Description

> Closes #85

Add a watch for secrets in the Gateway controller that will enqueue gateways if there's a change in the related cluster secret

## Verification steps

1. For testing purposes, as #52 is not done, update [this line of code](https://github.com/david-martin/multi-cluster-traffic-controller/blob/8bb9849ad642abdc847d5995ea34f3d07a187b17/pkg/controllers/gateway/gateway_controller.go#L337) to the following:
    ```go
    return []string{"mctc-workload-1"}
    ```

1. Run local-setup and start the controller

1. Create a sample gateway

1. Update the secret for the workload cluster

1. Verify that the sample gateway is reconciled
